source/elements/oneDAL/source/data_management/data_sources.rst

```diff
@@ -1,7 +1,117 @@
+.. highlight:: cpp
+.. default-domain:: cpp
+
 .. _data-sources:
 
 ============
 Data Sources
 ============
 
-TBD
+This section describes the types related to the :txtref:`data-source` concept.
+
+---------------
+CSV data source
+---------------
+Class ``csv::data_source`` is an api for accessing the data source represented as a :capterm:`csv file <CSV file>`.
+CSV data source can be used in :expr:`read` operation to extract data in text format from the given input file,
+process it using provided parameters (such as delimiter and parsing options), transform it into numerical representation,
+and store it as an in-memory :txtref:`dataset` of a chosen type.
+
+Supported type of in-memory dataset for :expr:`read` operation with CSV data source is :txtref:`table`.
+
+CSV data source requires input file name to be set in the constructor, while other parameters can be initialized with default values.
```

Comment:
while the other parameters of the constructor such as ... rely on default values

Reply:
Done.